app/src/main/java/com/myportfolio/mymovieapp/ui/CategoryFragment.kt

```diff
@@ -10,32 +10,42 @@ import androidx.fragment.app.activityViewModels
 import androidx.navigation.fragment.findNavController
 import androidx.recyclerview.widget.RecyclerView
 import com.myportfolio.mymovieapp.R
+import com.myportfolio.mymovieapp.databinding.FragmentCategoryBinding
+import com.myportfolio.mymovieapp.databinding.FragmentHomeBinding
 
 /**
  * A simple [Fragment] subclass.
  */
 class CategoryFragment : Fragment() {
 
+    private var binding: FragmentCategoryBinding? = null
```

Comment:
to set up the binding class, I recommend using a backing property as shown in the doc: https://developer.android.com/topic/libraries/view-binding#fragments

in this case we have a mutable and nullable `_binding` variable to initialize and clear the binding value as needed, and another `binding` variable, read-only and non-nullable, with a custom getter to access the value of _binding:

```suggestion
    private var _binding: FragmentCategoryBinding? = null
    private val binding get() = _binding!!
```

the read-only `binding` variable is valid between onCreateView and onDestroyView, and allows you to access the views without having to repeat the use of null checks (`?`) or not-null assertion (`!!`)

Reply:
this comment is valid to the other fragments as well